netaddr.go

```diff
@@ -179,24 +179,55 @@ func MustParseIP(s string) IP {
 // net.ParseIP at the moment, so returning ok=false from here just
 // means it'll try it a slower way.
 func parseIPv4(s string) (ip IP, ok bool) {
-	if len(s) < 7 || len(s) > 15 || strings.Count(s, ".") != 3 {
-		return ip, false
-	}
-	var v [4]uint64
-	for i := range v {
-		var numStr string
-		if dot := strings.IndexByte(s, '.'); dot == -1 {
-			numStr = s
-		} else {
-			numStr, s = s[:dot], s[dot+1:]
+	// Parse directly into the return value. This shaves another ~5%
```

Comment:
I'd rather pay that 5% cost.

Reply:
I assume you'd prefer that the resulting IP always be zero if !ok?

Reply:
I dislike this focus on always using the base constructors even when they make the computer's job harder, but under duress, done. Benchmark is now 20ns/IP instead of 19ns/IP.

Reply:
@mdlayher my preference is to always return a zero type on failure, because otherwise we open ourselves up to someone not checking the error and ending up using a half-built IP address.

It's moot with the change back to IPv4() though, since the return value no longer has intermediate state in it.